pkg/driver/node_test.go

```diff
@@ -2405,6 +2406,22 @@ func TestRemoveNotReadyTaint(t *testing.T) {
 	}
 }
 
+var mockRemovalFuncCalled = 0
+
+func mockRemovalFunc(_ cloud.KubernetesAPIClient) error {
+	mockRemovalFuncCalled += 1
+	if mockRemovalFuncCalled == 3 {
+		return nil
+	} else {
+		return fmt.Errorf("Taint removal failed!")
+	}
+}
+
+func TestRemoveTaintInBackground(t *testing.T) {
+	removeTaintInBackground(nil, mockRemovalFunc)
+	assert.Equal(t, mockRemovalFuncCalled, 3)
+}
+
```

Comment:
Suggest encapsulating both `mockRemovalFuncCalled` and `mockRemovalFunc` within `TestRemoveTaintInBackground` using an [anonymous function](https://gobyexample.com/closures) to remove the global variable:

```
func TestRemoveTaintInBackground(t *testing.T) {
	var mockRemovalFuncCalled int

	mockRemovalFunc := func(_ cloud.KubernetesAPIClient) error {
		mockRemovalFuncCalled += 1
		if mockRemovalFuncCalled == 3 {
			return nil
		} else {
			return fmt.Errorf("Taint removal failed!")
		}
	}

	removeTaintInBackground(nil, mockRemovalFunc)
	assert.Equal(t, mockRemovalFuncCalled, 3)
}
```

Reply:
Fixed in next version